src/sagemaker/jumpstart/cache.py

```diff
@@ -129,15 +149,44 @@ def get_region(self) -> str:
         """Return region for cache."""
         return self._region
 
-    def set_manifest_file_s3_key(self, key: str) -> None:
-        """Set manifest file s3 key. Clears cache after new key is set."""
-        if key != self._manifest_file_s3_key:
-            self._manifest_file_s3_key = key
+    def set_manifest_file_s3_key(
+        self,
+        key: str,
+        file_type: JumpStartS3FileType = JumpStartS3FileType.OPEN_WEIGHT_MANIFEST,
+    ) -> None:
+        """Set manifest file s3 key, clear cache after new key is set.
+
+        Raises:
+            ValueError: if the file type is not recognized
+        """
+        file_mapping = {
+            JumpStartS3FileType.OPEN_WEIGHT_MANIFEST: self._manifest_file_s3_key,
+            JumpStartS3FileType.PROPRIETARY_MANIFEST: self._proprietary_manifest_s3_key,
+        }
+        property_name = file_mapping.get(file_type)
+        if not property_name:
+            raise ValueError(
+                f"Bad value when setting manifest '{file_type}': must be in"
+                f" {JumpStartS3FileType.OPEN_WEIGHT_MANIFEST}"
```

Comment:
maybe do `error_message  = f"Bad value when setting manifest '{file_type}': must be in {' '.join([e.name for e in JumpStartS3FileType]}"` and then reuse everywhere